cin_validator/rules/cin2022_23/rule_8736.py

```diff
@@ -0,0 +1,188 @@
+from typing import Mapping
+
+import pandas as pd
+
+from cin_validator.rule_engine import CINTable, RuleContext, rule_definition
+from cin_validator.test_engine import run_rule
+from cin_validator.utils import make_census_period
+
+# Get tables and columns of interest from the CINTable object defined in rule_engine/__api.py
+
+Assessments = CINTable.Assessments
+AssessmentAuthorisationDate = Assessments.AssessmentAuthorisationDate
+AssessmentActualStartDate = Assessments.AssessmentActualStartDate
+LAchildID = Assessments.LAchildID
+# CINdetailsID = CINTable.Assessments
+
+
+Header = CINTable.Header
+ReferenceDate = Header.ReferenceDate
+
+# define characteristics of rule
+@rule_definition(
+    code=8736,
+    module=CINTable.Assessments,
+    message="For an Assessment that has not been completed, the start date must fall within the census year",
+    affected_fields=[AssessmentAuthorisationDate, AssessmentActualStartDate],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    # PREPARING DATA
+
+    # Replace ChildIdentifiers with the name of the table you need.
+    df = data_container[Assessments]
+    header = data_container[Header]
+    ref_date_series = header[ReferenceDate]
+    collection_start, collection_end = make_census_period(ref_date_series)
+
+    # Before you begin, rename the index so that the initial row positions can be kept intact.
+    df.index.name = "ROW_ID"
+
+    # lOGIC
+    # Implement rule logic as described by the Github issue.
+    # Put the description as a comment above the implementation as shown.
+
+    # Where present, if an <Assessments> group does not contain the <AssessmentAuthorisationDate> (N00160) then the <AssessmentActualStartDate> (N00159) must be on or between [Start_Of_Census_Year] and <ReferenceDate> (N00603)
+    condition_1 = df[AssessmentAuthorisationDate].isna
+    condition_2 = (df[AssessmentActualStartDate] >= collection_start) & (
+        df[AssessmentActualStartDate] <= collection_end
+    )
+
+    # get all the data that fits the failing condition. Reset the index so that ROW_ID now becomes a column of df
+    df_issues = df[condition_1 & condition_2].reset_index()
```

Comment:
With the way you've set up condition 2, this filter should either have a ~ before condition_2 as it's a not, or you'd need to change condition_2 itself to return rows that are not in the census period, as currently they return rows which are, according to the rule, correct.